xbmc/interfaces/legacy/ListItem.cpp

```diff
@@ -396,6 +398,21 @@ namespace XBMCAddon
           else if (key == "dateadded")
             item->GetVideoInfoTag()->m_dateAdded.SetFromDBDateTime(value.c_str());
         }
+
+        // For backward compatibility.
+        // FIXME: Remove this behaviour. It should be possible to set only tvshowtitle without
+        // having mediatype implicitly changed.
+        if (item->GetVideoInfoTag()->m_type == MediaTypeNone)
+        {
+          if (!item->GetVideoInfoTag()->m_strShowTitle.empty() && item->GetVideoInfoTag()->m_iSeason == -1)
+          {
+            item->GetVideoInfoTag()->m_type = MediaTypeTvShow;
+          }
+          else if (!item->GetVideoInfoTag()->m_artist.empty())
+          {
+            item->GetVideoInfoTag()->m_type = MediaTypeMusicVideo;
+          }
```

Comment:
Well since it's not used in the cases you change here it's not really an API change. I just prefer to do all the work so that we don't suddenly run into a case where we expect `CVideoInfoTag::m_type` to be `MediaTypeMovie` because we assumed that this legacy handling here takes care of it but it doesn't.